### PR TITLE
fix: hiding agent status on stopped workspaces

### LINKED ISSUE #3484: UI shows agent "connecting" when workspace is stopped
Screen Shot 2022-08-12 at 6 50 17 AM

Minor nit that Coder knows the workspace is stopped, so displaying the agent status is a bit moot. The green "Connecting" also implies that there's something working in the background. I think a better display would either to omit that or match the workspace Stopped display.

### PR DESCRIPTION
**Pull request by Kira-Pilot**
resolves #3484

We have an issue where workspace agent statuses are 'connecting' after a workspace has stopped. After chatting with @bpmct, we have decided to hide the agent status on stopped workspaces. We have also created tickets to address the agent status on the BE (see #3510 and #3511). This PR addresses hiding the status in the UI, only.

## Before
Screen Shot 2022-08-15 at 11 52 16 AM

## After
Screen Shot 2022-08-15 at 12 38 53 PM

### CODE REVIEW

File: site/src/util/workspace.ts [hunk 1]
```diff
@@ -6,6 +6,21 @@ import * as TypesGen from "../api/typesGenerated"
...
+  stopped = "Stopped",
+  canceling = "Canceling",
+  canceled = "Canceled",
+  deleting = "Deleting",
+  deleted = "Deleted",
+  queued = "Queued",
+  error = "Error",
+  loading = "Loading",
+}
+
```
Comment (Kira-Pilot) [inline site/src/util/workspace.ts:23]: I moved this out of a component directory and into our util file so we can have properly typed statuses.

File: site/src/components/Resources/Resources.tsx [hunk 2]
```diff
@@ -102,9 +106,11 @@ export const Resources: FC<ResourcesProps> = ({
                         {agent.name}
                         <div className={styles.agentInfo}>
                           <span className={styles.operatingSystem}>{agent.operating_system}</span>
-                          <span style={{ color: agentStatus.color }} className={styles.status}>
-                            {agentStatus.status}
-                          </span>
+                          {WorkspaceStateEnum[workspaceStatus] !== "Stopped" && (
```
Comment (jsjoeio) [inline site/src/components/Resources/Resources.tsx:109]: what if we used the Enum to get the value?
```suggestion
                          {WorkspaceStateEnum[workspaceStatus] !== WorkspaceStateEnum["stopped"] && (
```

**Review by kylecarbs — APPROVED** [review]
Looks good to me! Sorry for the slow review!

File: site/src/components/Resources/Resources.tsx (hunk 2 above)
Comment (Kira-Pilot) [reply site/src/components/Resources/Resources.tsx:109]: oh yeah hahaha thanks!

File: site/src/components/Resources/Resources.tsx [hunk 3]
```diff
@@ -102,9 +106,11 @@ export const Resources: FC<ResourcesProps> = ({
                         {agent.name}
                         <div className={styles.agentInfo}>
                           <span className={styles.operatingSystem}>{agent.operating_system}</span>
-                          <span style={{ color: agentStatus.color }} className={styles.status}>
-                            {agentStatus.status}
-                          </span>
+                          {WorkspaceStateEnum[workspaceStatus] !== WorkspaceStateEnum["stopped"] && (
+                            <span style={{ color: agentStatus.color }} className={styles.status}>
```
Comment (Kira-Pilot) [inline site/src/components/Resources/Resources.tsx:110]: @jsjoeio I just did that to get you to admit you like enums ;)

Comment (jsjoeio) [reply site/src/components/Resources/Resources.tsx:110]: Dang it...I didn't see that one coming 😂

The Battle of Enums
Kira: 1
Joe: 0

**Pull request merged**